File: frontend/sky-take-out/src/api/admin/competition.js
import request from '../../utils/request'

export const getCompetitionPageAPI = (params) => {
  const queryParams = {
    page: params.current || params.page || 1,
    pageSize: params.size || params.pageSize || 10,
    name: params.name
  }
  return request.get('/admin/competition/selectPage', { params: queryParams })
}

export const addCompetitionAPI = (data) => {
  return request.post('/admin/competition', data)
}

export const addCompetitionWithPhotoAPI = (data, file) => {
  const formData = new FormData()
  formData.append('name', data.name)
  formData.append('title', data.title || '')
  formData.append('description', data.description || '')
  formData.append('url', data.url || '')
  formData.append('date', data.date || '')
  formData.append('organizer', data.organizer || '')
  formData.append('location', data.location || '')
  formData.append('participantLimit', data.participantLimit || '')
  formData.append('prize', data.prize || '')
  formData.append('contact', data.contact || '')
  if (file) {
    formData.append('photo', file)
  }
  return request.post('/admin/competition/saveWithPhoto', formData, {
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export const updateCompetitionAPI = (data) => {
  return request.put('/admin/competition', data)
}

export const updateCompetitionWithPhotoAPI = (data, file) => {
  const formData = new FormData()
  formData.append('id', data.id)
  formData.append('name', data.name)
  formData.append('title', data.title || '')
  formData.append('description', data.description || '')
  formData.append('url', data.url || '')
  formData.append('date', data.date || '')
  formData.append('organizer', data.organizer || '')
  formData.append('location', data.location || '')
  formData.append('participantLimit', data.participantLimit || '')
  formData.append('prize', data.prize || '')
  formData.append('contact', data.contact || '')
  if (file) {
    formData.append('photo', file)
  }
  return request.put('/admin/competition/updateWithPhoto', formData, {
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export const deleteCompetitionAPI = (id) => {
  return request.delete(`/admin/competition/${id}`)
}

File: backend/sky-take-out/sky-pojo/src/main/java/com/sky/entity/Competition.java
package com.sky.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Competition implements Serializable {
    private static final long serialVersionUID = 1L;
    private Long id;//主键
    private String name;//比赛名称
    private String title;//比赛分类
    private String description;//描述
    private String url;//官网地址
    private String date;//举办时间
    private LocalDateTime updateTime;
    private String photourl;
    private LocalDateTime createTime;
    private String organizer;//主办方
    private String location;//举办地点
    private Integer participantLimit;//参赛人数限制
    private String prize;//奖项设置
    private String contact;//联系方式

    private Integer careCount;// 关注人数（非数据库字段）
}
